library/Versions.re

```diff
@@ -145,3 +146,20 @@ let getRemoteVersions = () => {
      )
   |> Lwt.return;
 };
+
+let throwIfInstalled = versionName => {
+  switch(getInstalledVersions()) {
```

Comment:
Instead of switching, you can also fold it to a value - an error is like an empty array in this case:

```reason
getInstalledVersions() |> Result.fold(_ => [||], xs => xs)
```

After folding, the code will probably be much cleaner 😄

Reply:
Folding is weird for newcomers, I hope that I got it right too. What it means:

```reason
let fold = (error, ok, result) =>
  switch (result) {
  | Error(x) => error(x)
  | Ok(x) => ok(x)
  }
```

It helps us take the wrapped value (the array) and the error, and convert it into one data structure. In this case, we don't care about the errors - we can assume it's an empty array, so folding it to an array makes sense, and we even remove one level of the pattern matching!